### PR TITLE
Add new Redfish config module to handle Dell-specific OEM extensions

### PR DESCRIPTION
**Pull request by jose-delarosa**
- Extends the existing RedfishUtils class and leverages its methods

##### SUMMARY

Implements code taken out by https://github.com/ansible/ansible/pull/51898 and provides module to implement all iDRAC-specific configuration settings. 

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

idrac_redfish_config

##### ADDITIONAL INFORMATION

This new module can be used as a model for other OEMs to implement their own Redfish extensions.

### CODE REVIEW

**Comment by jamescassell** [comment]
Is it possible to have it be a vendor-neutral redfish module?

**Comment by jose-delarosa** [comment]
@jamescassell There are already vendor-neutral modules in this same directory. This Dell-specific module implements code that was taken out of those vendor-neutral modules, and it follows a framework we discussed to implement OEM-specific Redfish implementations. Refer to https://github.com/ansible/ansible/issues/48364#issuecomment-439129431.

**Review by billdodd — COMMENTED** [review]
Since the 'timeout' PR has now been merged, this PR will need to be updated to add the timeout support. See the recent timeout related commit to idrac_redfish_command.py.

**Comment by jose-delarosa** [comment]
Yeah I realized that yesterday, will update today, thanks Bill.

**Review by billdodd — APPROVED** [review]
Looks good.

shipit

**Comment by jose-delarosa** [comment]
shipit

File: lib/ansible/modules/remote_management/redfish/idrac_redfish_config.py [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+  password:
+    required: true
+    description:
+      - Password for authentication with iDRAC
+  manager_attribute_name:
+    required: false
+    description:
+      - name of iDRAC attribute to update
+    default: 'null'
+    version_added: "2.8"
```
Comment (gundalow) [inline lib/ansible/modules/remote_management/redfish/idrac_redfish_config.py]:
```suggestion
```

File: lib/ansible/modules/remote_management/redfish/idrac_redfish_config.py [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+  manager_attribute_value:
+    required: false
+    description:
+      - value of iDRAC attribute to update
+    default: 'null'
+    version_added: "2.8"
```
Comment (gundalow) [inline lib/ansible/modules/remote_management/redfish/idrac_redfish_config.py]:
```suggestion
```

File: lib/ansible/modules/remote_management/redfish/idrac_redfish_config.py [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+  timeout:
+    description:
+      - Timeout in seconds for URL requests to iDRAC controller
+    default: 10
+    type: int
+    version_added: '2.8'
```
Comment (gundalow) [inline lib/ansible/modules/remote_management/redfish/idrac_redfish_config.py]:
```suggestion
```

File: lib/ansible/modules/remote_management/redfish/idrac_redfish_config.py [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'metadata_version': '1.1'}
+
+DOCUMENTATION = '''
+---
+module: idrac_redfish_config
+version_added: "2.8"
```
Comment (gundalow) [inline lib/ansible/modules/remote_management/redfish/idrac_redfish_config.py]:
```suggestion
version_added: '2.8'
```

File: lib/ansible/modules/remote_management/redfish/idrac_redfish_config.py [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+short_description: Manages servers through iDRAC using Dell Redfish APIs
+description:
+  - For use with Dell iDRAC operations that require Redfish OEM extensions
+  - Builds Redfish URIs locally and sends them to remote iDRAC controllers to
+    set or update a configuration attribute.
+options:
+  category:
+    required: true
+    description:
```
Comment (gundalow) [inline lib/ansible/modules/remote_management/redfish/idrac_redfish_config.py:27]:
```suggestion
    type: str
    description:
```

**Review by gundalow — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/remote_management/redfish/idrac_redfish_config.py (hunk 5 above)
Comment (jose-delarosa) [reply lib/ansible/modules/remote_management/redfish/idrac_redfish_config.py:27]: Updates committed. Thank you @gundalow

**Pull request merged**